paywall/src/__tests__/data-iframe/cache/localStorage.test.js

```diff
@@ -89,6 +97,58 @@ describe('localStorage cache', () => {
     })
   })
 
+  describe('getReadOnly', () => {
+    beforeEach(() => {
+      makeWindow()
+    })
+
+    it('value is not yet set', async () => {
+      expect.assertions(1)
+
+      expect(await getReadOnly(fakeWindow, 123, 'there')).toBeNull()
+    })
+
+    it('value is set', async () => {
+      expect.assertions(1)
+      fakeWindow.storage[storageId(123, nullAccount)] = JSON.stringify({
+        there: 'hello',
+      })
+
+      expect(await getReadOnly(fakeWindow, 123, 'there')).toEqual('hello')
+    })
+
+    it('value is set, but malformed', async () => {
```

Comment:
Is the setting of malformed data handled at a layer below the code in this pull request?

Reply:
localStorage is publicly accessible, so this guards against a malicious script trying to muck things up